planner/core/rule_aggregation_push_down.go

```diff
@@ -380,7 +380,7 @@ func (a *aggregationPushDownSolver) tryAggPushDownForUnion(union *LogicalUnionAl
 		}
 		newChildren = append(newChildren, newChild)
 	}
-	union.SetSchema(expression.NewSchema(newChildren[0].Schema().Columns...))
```

Comment:
The KeyInfo is also cloned if we use `Schema.Clone()`.
I think it's not expected here.